File: accommodation-booking/server/src/Services/Accomodation/src/main/java/accomodation/integration/eventBus/IEventBus.java
package accomodation.integration.eventBus;

import accomodation.integration.IIntegrationEventHandler;
import accomodation.integration.IntegrationEvent;
import org.springframework.stereotype.Service;

@Service
public interface IEventBus {
    <T extends IntegrationEvent, TH  extends IIntegrationEventHandler> void subscribe(Class<T> eventType, Class<TH> eventHandler);
    void publish(IntegrationEvent event);
}

File: accommodation-booking/server/src/EventBus.NET/EventBus/IEventBus.cs

namespace EventBus.NET.Integration.EventBus
{
    public interface IEventBus
    {
        void Subscribe<T, TH>()
            where T : IntegrationEvent
            where TH : IIntegrationEventHandler<T>;
        void Publish(IntegrationEvent @event);
    }
}
